Fixed build to work with R14,R15,R16,17 and 18

djustinek opened the pull request:
Fixed build scripts and code so it works with latest Erlang/OPT releases.

Commit:
Fixed build for Erlang 18.0

- Added Erlang version check and correct type selection
- Added erlang:timestamp() check script and correct compilation path for rebar and Makefile
- Added macro for TIMESTAMP and replaced direct calls with macros

Commit:
Added 18.0 to Travis tests

Commit:
Updated Travis config.

Commit:
Fixed tests compilation.

howleysv commented:
Good job!

I propose a change to move the conditional compilation into the .erl files to avoid issues in projects that include the emysql header.

See my patch: howleysv/Emysql@dda35e151917c0f43728c15aa4eee98f5aa6e1d3

Commit:
Applied patch https://github.com/howleysv/Emysql/commit/dda35e151917c0f43728c15aa4eee98f5aa6e1d3

Moved conditional compilation into .erl files to avoid issues in projects that include emysql header.

Commit:
Reverted to original state.

Builds failed on Erlang prior to 17

djustinek commented:
@howleysv : Proposed patch breaks compilation on pre-17 Erlang versions.
(https://travis-ci.org/Eonblast/Emysql/builds/77387690)

howleysv commented:
Sorry, I should have tested this myself!

I forgot that the built-in types could not be redefined in <17. Renaming to t_type as before seems to work (tested in 18.0 and R16B03-1):

howleysv/Emysql@00d2f864d8cea44a7260d1fccf480a8885e16455

Commit:
Moved conditional compilation into the .erl files.

Included patches:
- https://github.com/howleysv/Emysql/commit/dda35e151917c0f43728c15aa4eee98f5aa6e1d3
- https://github.com/howleysv/Emysql/commit/00d2f864d8cea44a7260d1fccf480a8885e16455

djustinek commented:
@howleysv, it's patched. Everything seems fine now.

nazar256 commented:
Why wouldn't you accept and apply this pull-request? Is something wrong with it or this repository is abandoned?
